uaclient/contract.py

```diff
@@ -171,6 +173,84 @@ def report_machine_activity(self):
             machine_token["activityInfo"] = response
             self.cfg.write_cache("machine-token", machine_token)
 
+    def get_magic_attach_token_info(self, magic_token: str) -> Dict[str, Any]:
+        """Request magic attach token info.
+
+        When the magic token is registered, it will contain new fields
+        that will allow us to know that the attach process can proceed
+        """
+        headers = self.headers()
+        headers.update({"Authorization": "Bearer {}".format(magic_token)})
+        url = API_V1_MAGIC_ATTACH
+
+        try:
+            response, _ = self.request_url(url, headers=headers)
+        except exceptions.UrlError as e:
+            if isinstance(e, exceptions.ContractAPIError):
+                if hasattr(e, "code"):
+                    if e.code == 401:
+                        raise exceptions.MagicAttachTokenError()
+                raise e
+
+            logging.exception(str(e))
+            raise exceptions.UserFacingError(
+                msg=messages.CONNECTIVITY_ERROR.msg,
+                msg_code=messages.CONNECTIVITY_ERROR.name,
+            )
+
+        return response
+
+    def new_magic_attach_token(self, email: str) -> Dict[str, Any]:
+        """Create a magic attach token for the user."""
+        headers = self.headers()
+        url = API_V1_MAGIC_ATTACH
+
+        try:
+            kwargs = {
+                "headers": headers,
+                "method": "POST",
+                "data": {"email": email},
+            }
+            response, _ = self.request_url(url, **kwargs)
```

Comment:
Why not pass the params directly?
```suggestion
        try:
            response, _ = self.request_url(
                API_V1_MAGIC_ATTACH,
                headers=headers,
                method="POST",
                data={"email": email},
            )
```